crates/router/src/connector/bluesnap.rs

```diff
@@ -702,12 +703,16 @@ impl api::IncomingWebhook for Bluesnap {
     fn get_webhook_object_reference_id(
         &self,
         request: &api::IncomingWebhookRequestDetails<'_>,
-    ) -> CustomResult<String, errors::ConnectorError> {
+    ) -> CustomResult<ObjectReferenceId, errors::ConnectorError> {
         let webhook_body: bluesnap::BluesnapWebhookBody =
             serde_urlencoded::from_bytes(request.body)
                 .into_report()
                 .change_context(errors::ConnectorError::WebhookSignatureNotFound)?;
-        Ok(webhook_body.reference_number)
+        Ok(ObjectReferenceId::PaymentId(
+            api_models::payments::PaymentIdType::ConnectorTransactionId(
+                webhook_body.reference_number,
+            ),
+        ))
```

Comment:
Need logic to detect type of webhook and return appropriate ObjectReferenceId as opposed to always returning PaymentId

Reply:
Refund Flow is not yet implemented for this connector